projects/rpx-xui-translation/src/lib/rpx-translate.pipe.ts

```diff
@@ -1,37 +1,49 @@
 import { AsyncPipe } from '@angular/common';
 import { ChangeDetectorRef, Pipe, PipeTransform } from '@angular/core';
 import { Observable } from 'rxjs';
-import { YesOrNoValue } from './rpx-language.enum';
 import { Replacements, RpxTranslationService } from './rpx-translation.service';
+import { YesOrNoValue } from './rpx-language.enum';
+import { TranslatedData } from './models/translated-data.model';
+import { replacePlaceholders } from './helpers/replace-placeholders/replace-placeholders.helper';
+import { matchCase } from './helpers/match-case/match-case.helper';
 
 @Pipe({
   name: 'rpxTranslate',
   pure: false
 })
 export class RpxTranslatePipe extends AsyncPipe implements PipeTransform {
-
   constructor(
     private translationService: RpxTranslationService,
     ref: ChangeDetectorRef
   ) {
     super(ref);
   }
 
-  transform<T = string>(value: T, replacements?: Replacements | null, yesOrNo?: string): T | null {
+  transform<T = string>(value: T, replacements?: Replacements | null, yesOrNoValue?: string): T | null {
     if (typeof value === 'string') {
-      let o: Observable<string>;
-      if (replacements) {
-        o = this.translationService.getTranslationWithReplacements(value, replacements);
-      } else if (yesOrNo?.toLowerCase() === YesOrNoValue.YES.toLowerCase() || yesOrNo?.toLowerCase() === YesOrNoValue.NO.toLowerCase()) {
-        const yesOrNoValue = yesOrNo?.toLowerCase() === YesOrNoValue.YES.toLowerCase() ? YesOrNoValue.YES : YesOrNoValue.NO;
-        o = this.translationService.getYesOrNoTranslationReplacement(value, yesOrNoValue);
+      let retString: string;
+      const o: Observable<TranslatedData> = this.translationService.getTranslation(value);
```

Comment:
getTranslation should return a string. Within the pipe you should call the correct method on the service, as the methods on the service should only return Observable<string>. The fact that TranslatedData objects are used should remain internal to the service.

By having this logic in the service it is always available, without having to reproduce the logic you currently have in the pipe when you have to use it outside a template.

The only logic inside the pipe should be around using the parameters to determine which of the methods on the service to call.